okcupyd/json_search.py

```diff
@@ -70,15 +75,59 @@ def fetch_once(self, count=18):
 
 class SearchJSONFetcher(object):
 
-    def __init__(self, session=None, **options):
+    def __init__(self, session=None, order_by="MATCH", location=None, keywords=None, **options):
         self._session = session or Session.login()
-        self._options = options
+        self._order_by = order_by.upper()
+        if self._order_by not in ('MATCH', 'ONLINE', 'SPECIAL_BLEND'): # TODO: haven't verified that options other than 'MATCH' still work; the web interface no longer provides a way to modify this
+            raise TypeError
+        self._location = location # location string, haven't implemented
+        self._keywords = keywords # list or space-delimited string of words to match, haven't implemented
         self._parameters = search_filters.build(**options)
 
     def _query_params(self, after=None, count=None):
         search_parameters = {
             'after': after,
-            'limit': count
+            'limit': count,
+            # other admin parameters
+            'debug': 0,
+            'username': "",
+            'save_search': 1,
+            'order_by': self._order_by,
+
+            'i_want': None,
+            'they_want': None,
+            'tagOrder': [],
+
+            # optional parameters
+            'minimum_height': None,
+            'maximum_height': None,
+            'languages': 0,
+            'speaks_my_language': 0, # TODO shall we make 1 the default?
+            'ethnicity': [],
```

Comment:
These defaults should be a part of the filter class. We can just put an attribute on the Filter class that is checked if the decider is false that we can use to specify these values.